cmd/nodecmd/dynamic_ips.go

```diff
@@ -79,21 +84,49 @@ func getPublicIPForNodesWoEIP(nodesWoEIP []models.NodeConfig) (map[string]string
 	return publicIPMap, nil
 }
 
-func updateAnsiblePublicIPs(clusterName string) error {
+// update public IPs
+// - in ansible inventory file
+// - in host config file
+func updatePublicIPs(clusterName string) error {
 	clusterNodes, err := getClusterNodes(clusterName)
 	if err != nil {
 		return err
 	}
-	nodesWoEIP := getNodesWoEIPInAnsibleInventory(clusterNodes)
-	if len(nodesWoEIP) > 0 {
-		publicIP, err := getPublicIPForNodesWoEIP(nodesWoEIP)
+	nodesWithDynamicIP, err := getNodesWithDynamicIP(clusterNodes)
+	if err != nil {
+		return err
+	}
+	if len(nodesWithDynamicIP) > 0 {
+		nodeIDs := utils.Map(nodesWithDynamicIP, func(c models.NodeConfig) string { return c.NodeID })
+		ux.Logger.PrintToUser("Nodes with dynamic IPs in cluster: %s", nodeIDs)
+		publicIPMap, err := getPublicIPsForNodesWithDynamicIP(nodesWithDynamicIP)
 		if err != nil {
 			return err
 		}
-		err = ansible.UpdateInventoryHostPublicIP(app.GetAnsibleInventoryDirPath(clusterName), publicIP)
-		if err != nil {
+		changed := 0
+		for _, node := range nodesWithDynamicIP {
+			if node.ElasticIP != publicIPMap[node.NodeID] {
+				ux.Logger.PrintToUser("Updating IP information from %s to %s for node %s",
+					node.ElasticIP,
+					publicIPMap[node.NodeID],
+					node.NodeID,
+				)
+				changed++
+			}
+			node.ElasticIP = publicIPMap[node.NodeID]
+			if err := app.CreateNodeCloudConfigFile(node.NodeID, &node); err != nil { //nolint:gosec
+				return err
+			}
+		}
+		if changed == 0 {
+			ux.Logger.PrintToUser("All node IPs are the same. No update needed")
```

Comment:
`No changes to IPs detected` might be a better wording. Open for any other suggestions